packages/core/src/types/asyncapi3.ts

```diff
@@ -0,0 +1,1189 @@
+      'amqp1',
+      'mqtt',
+      'mqtt5',
+      'nats',
+      'jms',
+      'sns',
+      'solace',
+      'sqs',
+      'stomp',
+      'redis',
+      'mercure',
+      'ibmmq',
+      'googlepubsub',
+      'pulsar',
+    ];
+  },
+  additionalProperties: { type: 'object' },
+};
+
+const OperationBindings: NodeType = {
+  properties: {},
+  allowed() {
+    // allow all supported values, not all have deep linting
+    return [
+      'http',
+      'ws',
+      'kafka',
+      'anypointmq',
+      'amqp',
+      'amqp1',
+      'mqtt',
+      'mqtt5',
+      'nats',
+      'jms',
+      'sns',
+      'solace',
+      'sqs',
+      'stomp',
+      'redis',
+      'mercure',
+      'ibmmq',
+      'googlepubsub',
+      'pulsar',
+    ];
+  },
+  additionalProperties: { type: 'object' },
+};
+
+const OperationTrait: NodeType = {
+  properties: {
+    tags: 'TagList',
+    title: { type: 'string' },
+    summary: { type: 'string' },
+    description: { type: 'string' },
+    externalDocs: 'ExternalDocs',
+    security: 'SecuritySchemeList',
+
+    bindings: 'OperationBindings',
+  },
+  required: [],
+};
+
+const MessageTrait: NodeType = {
+  properties: {
+    headers: (value: unknown) => {
+      if (typeof value === 'function' || (typeof value === 'object' && !!value)) {
+        return {
+          properties: {
+            schema: 'Schema',
+            schemaFormat: { type: 'string' },
+          },
+        };
+      } else {
+        return 'Schema';
+      }
+    },
+    correlationId: 'CorrelationId',
+
+    contentType: { type: 'string' },
+    name: { type: 'string' },
+    title: { type: 'string' },
+    summary: { type: 'string' },
+    description: { type: 'string' },
+    tags: 'TagList',
+    externalDocs: 'ExternalDocs',
+    bindings: 'MessageBindings',
+    examples: 'MessageExampleList',
+  },
+  additionalProperties: {},
+};
+
+const Operation: NodeType = {
+  properties: {
+    action: { type: 'string', enum: ['send', 'receive'] },
+    channel: 'Channel',
+    title: { type: 'string' },
+    tags: 'TagList',
+    summary: { type: 'string' },
+    description: { type: 'string' },
+    externalDocs: 'ExternalDocs',
+    operationId: { type: 'string' },
+    security: 'SecuritySchemeList',
+
+    bindings: 'OperationBindings',
+    traits: 'OperationTraitList',
+    messages: 'MessageList',
+    reply: 'OperationReply',
+  },
+  required: ['action', 'channel'],
+};
+
+const OperationReply: NodeType = {
+  properties: {
+    channel: 'Channel',
+    messages: 'MessageList',
+    address: 'OperationReplyAddress',
+  },
+};
+
+const OperationReplyAddress: NodeType = {
+  properties: {
+    location: { type: 'string' },
+    description: { type: 'string' },
+  },
+  required: ['location'],
+};
+
+const MessageExample: NodeType = {
+  properties: {
+    payload: { isExample: true },
+    summary: { type: 'string' },
+    name: { type: 'string' },
+    headers: { type: 'object' },
+  },
+};
+
+const Schema: NodeType = {
```

Comment:
Could you reuse the 2.6 schema as it shares the [same draft](https://www.asyncapi.com/docs/reference/specification/v3.0.0#schemaObject) with 3.0?

Reply:
done. But also fixed types for `schema->items` for async2. It is related to the `bundle` command.

Reply:
`Schema` definition should use #1608 updates. 

originally, asyncapi.ts had the Schema object defined with the wrong JSON Schema draft. It was using 2020-12 definitions, but it should be draft-07.

Reply:
Thanks, @jeremyfiel, please merge your PR and I'll rebase it. I know about draft-07 and don't want to duplicate your PR, it'll produce conflict between our branches.